lib/backends/prometheus.ts

```diff
@@ -0,0 +1,161 @@
+import { AddressInfo, createServer } from "net";
+import http from "http";
+import express from "express";
+
+import client from "prom-client";
+import { ILogger } from "nchat-dev-common";
+
+import { DefaultBackendSettings } from "../constants";
+import { IBackendSettings, IMetricsBackend } from "../types";
+
+export class PrometheusMetricsBackend implements IMetricsBackend {
+    private _client: typeof client;
+    private _logger: ILogger;
+    private _metricsServer?: http.Server;
+    private _desiredServerPort: number;
+    private _serverPort: number;
+    private _defaultMetricsTimeout: number;
+    private app?: express.Express;
+
+    constructor(settings?: Partial<IBackendSettings>) {
+        const {
+            logger,
+            defaultMetricsTimeout,
+            serverPort,
+        } = { ...DefaultBackendSettings, ...settings };
+
+        this._setupServer();
+
+        this._client = client;
+        this._logger = logger || {
+            // tslint:disable-next-line: no-console
+            debug: () => console.debug,
+            // tslint:disable-next-line: no-console
+            info: () => console.log,
+            // tslint:disable-next-line: no-console
+            warn: () => console.warn,
+            // tslint:disable-next-line: no-console
+            error: () => console.error,
+            child: () => this._logger,
+            filterFunction: (x) => x,
+        };
+
+        this._desiredServerPort = serverPort || 0;
```

Comment:
We still need it to satisfy the transpiler, because of the `Partial` type of the constant: `const DefaultBackendSettings: Partial<IBackendSettings>`. If I omit the type, then it isn't needed, but IMO having it there is more readable and preferable.